docs (#272)
infinum/eightshift-docs: website/forms/php/helpers.md

---
id: helpers
title: Helpers
---

Eightshift Forms offers static helpers that can be implemented in your project. The key difference between these helpers and filters is that you can use the former in your theme or plugin code regardless of the load cycle.

### esFormsGetFormIdByName

> This function retrieves the custom unique name of a form as set in the settings when provided with the form ID. When applying specific settings to one or more forms using project filters, it is handy. Usually, forms are identified by their IDs, but if you switch the environment, the form ID may change. This helper helps you identify the form by its name instead of its ID.

Reviewer: ```suggestion
This function retrieves the custom, unique name of a form, as set in the Form settings, from the provided form ID. It's useful when applying specific settings to one or more forms using filters in your project.

 Form identification by name (instead of the default "by ID"), can be useful if switching environments when the form ID could change.
```

```php
if (\function_exists('esFormsGetFormIdByName')) {
  echo esFormsGetFormIdByName('<form-id>');
}
```

### esFormsDecryptor

> Decrypt and encrypt methods encode variations set in the form settings and passed to the thank you page upon form submission. This method lets you decode and use the variation in your theme or plugin.

Reviewer: ```suggestion
This helper provides decrypt and encrypt methods for variations set in the Form settings, passed to the "Thank you" page after form submission.
```
(this one was a bit _što je pisac htio reći_, so feel free to ignore it if I've missed the point 😄)

```php
if (\function_exists('esFormsDecryptor')) {
  $variation = isset($_GET['es-variation']) ? \esFormsDecryptor(\sanitize_text_field(\wp_unslash($_GET['es-variation']))) : '';
}
```

### esFormsGeolocationCountriesList

> The Geolocation countries list method generates a list of countries available in the geolocation API and country list feature.

Reviewer: ```suggestion
This method generates a list of countries available in the geolocation API and country fields.
```

```php
if (\function_exists('esFormsGeolocationCountriesList')) {
  $countriesList = esFormsGeolocationCountriesList();
}
```

### esFormsGetSelectOptionsArrayFromString

> Once the select block options are outputted, they will be stored in HTML format, making it difficult to manipulate the data. However, there is a method to convert the HTML string into an array, allowing you to use it in your theme or plugin more easily.

Reviewer: ```suggestion
This helper parses the block grammar output (that the Gutenberg editor generates) for block options, making it easy to use them in your projects.
```

```php
if (\function_exists('esFormsGetSelectOptionsArrayFromString')) {
  $select = esFormsGetSelectOptionsArrayFromString('<options>');
}
```

### esFormsGetComponentsRender

> This helper allows you to access the Eightshift Forms component render method, enabling you to utilize all our form components in your theme or plugin. If you already have a style set for a select component, you can use this helper to render the component and pass the necessary data to it for use in your project.

Reviewer: ```suggestion
This helper allows you to access the Eightshift Forms component `render` method, enabling you to utilize all Forms  components in your projects. If you already have a style set for the selected component, you can use this helper to render it with all the necessary data and styling applied.
```
